app/src/parser_impl.c

```diff
@@ -265,7 +265,7 @@ parser_error_t json_matchOptionalArray(parsed_json_t *parsedJson,
                                   uint16_t tokenIdx, uint16_t *valueTokenIdx) {
     CHECK_PARSER_ERR(json_validateToken(parsedJson, tokenIdx))
 
-    if (tokenIdx + 4 >= parsedJson->numberOfTokens) {
+    if (!(tokenIdx + 4 < parsedJson->numberOfTokens)) {
```

Comment:
Out of curiosity, what is the reason of switching from `<=` to `!>` ?

Reply:
I believe this way it is easier to understand in the context. It forced me to think for quite a while, but I though that I am just tired. Then ledger guys had questions here too.